pkg/sloop/ingress/kubewatcher.go

```diff
@@ -306,14 +306,26 @@ func (i *kubeWatcherImpl) processUpdate(kind string, obj interface{}, watchResul
 	glog.V(99).Infof("processUpdate: obj json: %v", resourceJson)
 
 	kubeMetadata, err := kubeextractor.ExtractMetadata(resourceJson)
+	EventInfo, err1 := kubeextractor.ExtractEventInfo(resourceJson)
+	InvolvedObject, err2 := kubeextractor.ExtractInvolvedObject(resourceJson)
 	if err != nil || kubeMetadata.Namespace == "" {
 		// We are only grabbing namespace here for a prometheus metric, so if metadata extract fails we just log and continue
 		glog.V(2).Infof("No namespace for resource: %v", err)
 	}
+	if err1 != nil {
+		glog.V(2).Infof("Extract event info: %v", err1)
+	}
+	if err2 != nil {
+		glog.V(2).Infof("Error occured while extracting Involved Object Info: %v", err2)
+	}
+	if enableGranularmetrics && kind == "Event" {
+		metricIngressGranularKubewatchcount.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace, kubeMetadata.Name, InvolvedObject.Kind, EventInfo.Reason, EventInfo.Type).Inc()
+		metricIngressGranularKubewatchbytes.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace, kubeMetadata.Name, InvolvedObject.Kind, EventInfo.Reason, EventInfo.Type).Add(float64(len(resourceJson)))
+	}
 	metricIngressKubewatchcount.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace).Inc()
 	metricIngressKubewatchbytes.WithLabelValues(kind, watchResult.WatchType.String(), kubeMetadata.Namespace).Add(float64(len(resourceJson)))
 
-	glog.V(common.GlogVerbose).Infof("Informer update (%s) - Name: %s, Namespace: %s, ResourceVersion: %s", watchResult.WatchType, kubeMetadata.Name, kubeMetadata.Namespace, kubeMetadata.ResourceVersion)
+	glog.V(common.GlogVerbose).Infof("Informer update (%s) - Name: %s, Namespace: %s, ResourceVersion: %s, Reason: %s, Type: %s", watchResult.WatchType, kubeMetadata.Name, kubeMetadata.Namespace, kubeMetadata.ResourceVersion, EventInfo.Reason, EventInfo.Type)
```

Comment:
This change should also be in the if condition and the original glog should be in false.

Reply:
Okay got it. Will do that

Reply:
Noted